scRNA-seq-advanced/02-celltype_assignment.Rmd

```diff
@@ -315,185 +316,324 @@ While this technique might be good for some simple experiments, and can be usefu
+We seem to have a pretty good set of assignments, with most of the cell type assignments falling into groupings consistent with what we see in the UMAP plot.
 
-This is a pretty clean plot, for which we can thank that this is a PBMC data set.
+We can thank the fact that this is a PBMC sample and that we have a good reference data set for these cell types for the cleanliness of this plot.
+Quite often with other kinds of samples (especially cancer cells!) things will be much less clean!
 
-```{r}
+We can also look to see how the cell type assignments are distributed using the base R function `table()`
+Since we like to keep track of the cells that ended up as `NA` in the pruned labels, we will include the `useNA = "ifany"` argument.
+
+```{r cell type table}
 table(singler_result$pruned.labels, useNA = "ifany")
 ```
 
- 
-We can do some cleanup using `forcats`.
+### Exploring finer labels
 
-```{r}
-# combine all of the uncommon cell types
-common_labels <- singler_result$pruned.labels %>%
-  forcats::fct_lump_min(100)
-sce$celltype_common <- common_labels
-```
+In the previous cell typing, we used the `label.main` column, but we also had `label.fine`, so let's use that to explore the data set in a bit more detail. 
 
+We will also take this time to dive a bit deeper into the steps that `SingleR` performed. 
+As mentioned, the first step is training the model, during which we identify the genes that will be used for the correlation analysis later.
+While this step is not particularly slow, if we were classifying multiple samples, we would not want to have to repeat it for every sample.
 
-```{r}
-scater::plotUMAP(sce, colour_by = "celltype_common")
-```
+To do the training, we will use the `trainSingleR()` function. 
+For this we will start with our reference and the labels we want to train the model with.
 
-We can also use the finer-grained cell type data.
+We can then specify the method used to select the genes that will be used for classification.
+The default method is `"de"`, which performs a differential expression analysis for each pair of labels, but we could also use `"sd"` to select the genes which are most variable across labels, or `"all"` to use all genes.
+If we want to get really fancy, we could even provide a specific list of genes to use.
 
-If we were classifying multiple samples, we would want to do the training only once!
+We should note here that the reference dataset for `SingleR` does not need to be from a compendium like `celldex`!
+If you have any well-classified data set that you want to use as a reference, you can, as long as you can create a gene by sample expression matrix and a vector of cell types for each sample. 
+You will want to ensure that the cell types you expect to see are present in the reference data set, and data should be normalized, but otherwise the method can be quite flexible. 
```

Comment:
```suggestion
You will want to ensure that the cell types you expect to see in your sample are present in the reference dataset, and data should be normalized, but otherwise the method can be quite flexible. 
```
I'm changing to `dataset` throughout because in the beginning that's what you were using. Either one is fine, but I think we just want to be consistent.